src/Compilers/CSharp/Portable/CSharpFileSystemExtensions.cs

```diff
@@ -28,6 +28,9 @@ public static class CSharpFileSystemExtensions
         /// <exception cref="ArgumentNullException">Compilation or path is null.</exception>
         /// <exception cref="ArgumentException">Path is empty or invalid.</exception>
         /// <exception cref="IOException">An error occurred while reading or writing a file.</exception>
+#if NET6_0_OR_GREATER
+        [System.Diagnostics.CodeAnalysis.RequiresUnreferencedCode(TrimWarningMessages.NativePdbsNotSupported)]
+#endif
```

Comment:
Really not a fan of this `#if` pattern getting sprinkled around everywhere. Can we instead conditionally define `RequiresUnreferencedCode` as a polyfill? 

```cs
#if !NET6_0_OR_GREATER
namespace System.Diagnostics.CodeAnalysis;

internal sealed class ReqiresUnreferencedCodeAttribute : Attribute 
{
  ...
}
#endif
```

Then we can just use `RequiresUnreferencedCode` without a slew of `#if` in our code base.

Reply:
I can do the polyfill thing.